tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/bir/BirVariableOptimizationTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.test.bir;
+
+import org.ballerinalang.test.util.BCompileUtil;
+import org.ballerinalang.test.util.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.ballerinalang.compiler.bir.emit.BIREmitter;
+import org.wso2.ballerinalang.compiler.bir.model.BIRNode;
+import org.wso2.ballerinalang.compiler.bir.optimizer.LivenessAnalyzer;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+/**
+ * Test to confirm the functionality of the {@link LivenessAnalyzer}.
+ */
+public class BirVariableOptimizationTest {
+    private final BIREmitter birEmitter;
+    private CompileResult result;
+
+    public BirVariableOptimizationTest() {
+        birEmitter = BIREmitter.getInstance(new CompilerContext());
+        result = BCompileUtil.compileAndGetBIR("test-src/bir/biroptimizer.bal");
+    }
+
+    @Test(description = "Test the liveness analysis on functions")
+    public void testFunctions() {
+        ((BLangPackage) result.getAST()).symbol.bir.functions.forEach(this::assertFunctions);
+    }
+
+    @Test(description = "Test the liveness analysis on attached functions")
+    public void testAttachedFunctions() {
+        ((BLangPackage) result.getAST()).symbol.bir.typeDefs.forEach(
+                typeDefinition -> typeDefinition.attachedFuncs.forEach(this::assertFunctions));
+    }
+
+    private void assertFunctions(BIRNode.BIRFunction func) {
+        String expectedBir = null;
+        try {
+            expectedBir = readFile(func.getName().getValue());
+        } catch (IOException e) {
+            Assert.fail("Failed when reading file", e);
+        }
+        if (!"".equals(expectedBir)) {
```

Comment:
So if the expectedBir is an empty string, then the test case would pass here?

Reply:
yes. The file is not in the folder. We test for files which are in the folder. For `...init` or `...start` or any such functions we chose to ignore there won't be a file for that. Then we ignore. If the file is empty also it will ignore I guess.

Reply:
ack